framework/decode/file_processor.h

```diff
@@ -193,6 +193,54 @@ class FileProcessor
     }
 
   private:
+    // Must be define before the Seek calls below
+    class ActiveFiles
+    {
+      public:
+        class Ref
+        {
+          public:
+            ~Ref();
+            Ref(const Ref&) = delete;
+            Ref(Ref&&)      = delete;
+
+            std::string GetFilename() const;
+            FILE*       GetFile() const;
+            bool        FileSeek(int64_t offset, util::platform::FileSeekOrigin origin);
+
+          private:
+            ActiveFiles& active_file;
+            friend class ActiveFiles;
+            Ref(ActiveFiles& active_file_);
+        };
+        using RefPtr = std::unique_ptr<Ref>;
```

Comment:
Can't we just use a simple pointer or reference instead of this new class and the `unique_ptr`.

From https://en.cppreference.com/w/cpp/container/unordered_map.html:
```
References and pointers to either key or data stored in the container are only invalidated
by erasing that element, even when the corresponding iterator is invalidated.
```

So if I understand this correctly pointers or references are safe and should work for our case.
`SetActiveFile` functions then can just `file_stack_.emplace_back(file_entry->second, execute_till_eof);` and this new class will not be necessary. `file_stack_.pop_back()` should take care of freeing resources without the need of smart pointers